services/src/machine_learning/name.rs

```diff
@@ -0,0 +1,154 @@
+use postgres_types::{FromSql, ToSql};
+use serde::{de::Visitor, Deserialize, Serialize};
+use utoipa::ToSchema;
+
+const NAME_DELIMITER: char = ':';
+
+#[derive(Debug, Clone, Hash, Eq, PartialEq, FromSql, ToSql)]
+pub struct MlModelName {
+    pub namespace: Option<String>,
+    pub name: String,
+}
+
+impl MlModelName {
+    /// Canonicalize a name that reflects the system namespace and model.
+    fn canonicalize<S: Into<String> + PartialEq<&'static str>>(
+        name: S,
+        system_name: &'static str,
+    ) -> Option<String> {
+        if name == system_name {
+            None
+        } else {
+            Some(name.into())
+        }
+    }
+
+    pub fn new<S: Into<String>>(namespace: Option<String>, name: S) -> Self {
+        Self {
+            namespace,
+            name: name.into(),
+        }
+    }
+}
+
+impl From<MlModelName> for geoengine_datatypes::machine_learning::MlModelName {
+    fn from(name: MlModelName) -> Self {
+        Self {
+            namespace: name.namespace.clone(),
```

Comment:
why is that clone needed?

Reply:
I removed it 07374b850dd9a0d5638233a4a1df5deabb1dceea